### ISSUE TITLE
Instructions to get Lua dependencies outside Docker

### ISSUE THREAD

**Issue by jacopofar** [rustprooflabs/pgosm-flex#81]
Hello and thanks for the useful tool!

I'm trying to run the script on my machine (macOS 11.2), and when I start the script it complains about the missing dkjson package. I couldn't find instructions about installing Lua dependencies and don't know anything about this language, are there recommended ways to install the dependencies? [dkjson seems to be not mantained anymore](https://github.com/LuaDist/dkjson)

Eventually I did this:
* run `brew install luarocks`
* `luarocks install dkjson`
* `LUA_PATH="?;?.lua;/usr/local/share/lua/5.4/?.lua;$(pwd)/style?"; osm2pgsql --slim --drop \
    --output=flex --style=./run-all.lua [...]`

it works, but seems cumbersome and I think I'm missing something

**Comment by rustprooflabs** [comment]
Hi - Glad you are finding it helpful!  Thank you for the feedback.

I found and followed the notes from the [osm2pgsql example places.lua](https://github.com/openstreetmap/osm2pgsql/blob/master/flex-config/places.lua#L5-L7), which suggests `LuaRocks` as one of the options.  In my case I use `sudo apt install lua-dkjson` on Ubuntu and that seems to get everything on the path worked out. The Docker image [installs it here](https://github.com/rustprooflabs/pgosm-flex/blob/main/Dockerfile#L12).

The documentation is a bit lacking and out of order, I will work on improving that in the coming days. The README.md needs to document the pre-reqs including the `dkjson` requirement, currently that is not mentioned at all.  (Of course, PRs are welcome too!)

I am not sure what to do about that project being stale yet.  I used it because it was referenced from osm2pgsql and it seemed to work. There may be other options like a more active fork or newer alternative. I am also new to Lua and have not fully explored the options available.

**Comment by jacopofar** [comment]
Thanks!
My use case is to get a database (and the relative SQL dump) populated from a pbf file in an automated way, to be able to run tests easily.

Eventually I wrote these two scripts:

https://gist.github.com/jacopofar/5b554eb45e660d186ad71f9331a75834#file-sql_flex_dump_from_pbf-sh
https://gist.github.com/jacopofar/1e77101264c968738860808341076b25#file-database_from_pbf-sh

one is an abridged version of the one inside this repository, that simply invokes the import from a pbf file and uses pg_dump to produce the dump from the DB.

The second one uses the Docker image of pgosm-flex to run the first script, mounts the current folder to make the pbf file directly accessible to it and then reads the dump back into another container running a PostGIS 13 instance. This way the process is completely automated, and I get a DB that is ready to be used.

Since the DB is not immediately ready I used `pg_isready` to ensure it's waited for, but noticed there's a sudden restart of the server once it's ready first, so I have to wait twice.

The sql files are left in the `/tmp/schema_data folder`

**Comment by rustprooflabs** [comment]
I overhauled the documentation for this project, the prerequisites for Ubuntu are now documented under [MANUAL-STEPS-RUN.md](https://github.com/rustprooflabs/pgosm-flex/blob/main/MANUAL-STEPS-RUN.md), linked to from the main README.md.  I did not include the `brew install` step for `dkjson` since that's only one of of what I suspect are a number differences between Ubuntu and Mac OS.  I do not currently have a Mac so cannot find/verify differences in order to write accurate and complete instructions.

I adapted a check similar to your GIST using `pg_isready` to be part of the Docker runtime script. The double check is indeed required and not just for low memory environments, it seems to be part of the initial Pg setup when the container is created to bounce it the way it does.  Adding the time to sleep and wait (minimum 15 seconds, goes up in 5 sec intervals) is negligible in the overall scheme of loading OSM data.

I created #88 to not lose sight of the `dkjson` dependency being unmaintained.

I think that covers everything outlined in this issue, closing.  Let me know if I missed something!
